next.constants.mjs

```diff
@@ -32,9 +32,12 @@ export const ENABLE_STATIC_EXPORT =
  *
  * @see https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#framework-environment-variables
  */
-export const BASE_URL = process.env.NEXT_PUBLIC_VERCEL_URL
-  ? `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`
-  : process.env.NEXT_PUBLIC_BASE_URL || 'https://nodejs.org';
+export const BASE_URL =
+  process.env.VERCEL_ENV === 'production'
```

Comment:
I'm not sure if hardcoding the nodejs.org here is a good idea.

In the original issue we were thinking in simply giving higher priority to NEXT_PUBLIC_BASE_URL (which we can easily configure on our Vercel Dashboard)

TO L;DR swap the position of the NEXT_PUBLIC_VERCEL_URL with the BASE_URL one

Reply:
(Also we already have a constant for VERCEL_ENV)

Reply:
My PR doesn't require any configuration, but if you prefer to set an production env var as `NEXT_PUBLIC_BASE_URL=https://nodejs.org` that would work too.

Free free to close this PR if you already have a solution.

Reply:
I suppose another option is to add a `.env.production` file with `NEXT_PUBLIC_BASE_URL=https://nodejs.org`.

It would be assigned every time your run `next build` which would be for all Vercel deployments (both preview and production).

Do you want me to add that?

Reply:
Would that file also be served and available to the serverless functions?

I'm not much keen on .env files on the main repo, and simply let whoever that consumes this to configure it the way they want.

Hence why for me is more convenient setting up a env variable on Vercel itself.

But Im all ears 🙃

Reply:
I understand your PR initial changeset would fix the current symptom, and require no actionables, but it does in depriment of allowing any customisation of the base url on production vercel environments.

Reply:
> Would that file also be served and available to the serverless functions?

Yes, it works the same as normal env vars. 

Learn more: https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables

Reply:
I assumed this was a high priority PR since a broken sitemap means new pages might not be indexed by google and other search engines.

(the current sitemap urls have `X-Robots-Tag: noindex` since those are not using a production domain)

Reply:
Well, kinda yes. But that doesn't mean we can provide a future-proof solution given the nature and complexity of this; + setting an env variable on vercel is as fast as, something really fast (for the lack of words) 😅

Don't get me wrong, not criticising your approach, I'm simply making a review 🙃

Reply:
Ok sounds like you want the env var in the dashboard solution so I pushed [da7cb21](https://github.com/nodejs/nodejs.org/pull/5599/commits/da7cb21488f6dd6ae34fb0c5b20691f9130089a8)

Reply:
I mean, I'm fine with both solutions, I've just mentioned the variable one is convenient for me 🙃